### PR TITLE
Add map

### PR DESCRIPTION
**Pull request by cllns**
Uses [mapbox](https://www.mapbox.com/). My 'accessToken' is hard-coded into the JS. Instead, we could pull that in from environment variables, but the accessToken is going to be viewable to the public regardless, so I don't think it's worth it.

People can choose what level of 'accuracy' they want, which is how far the map is zoomed into a specific place, and how large the blue circle is. Eventually it'd be nice to load the map live next to the form, so the users can see what the circle looks like.

### CODE REVIEW

File: app/assets/stylesheets/partials/_posts.scss [hunk 1]
```diff
@@ -53,42 +53,41 @@
...
+    }
   }
-}
 
-.search-form {
-  @include span-columns(6);
-  margin-left: 0;
-  width: 100%;
-}
+  .search-form {
```
Comment (mi-wood) [inline app/assets/stylesheets/partials/_posts.scss]: any idea what the dark green highlighting is? I wasn't able to find a quick answer for this from github docs

File: app/views/posts/_form.html.erb [hunk 2]
```diff
@@ -81,6 +86,17 @@
   </p>
 
   <p>
+    <%= f.label :accuracy %><br>
```
Comment (mi-wood) [inline app/views/posts/_form.html.erb]: Should we add some sort of long description for this?

File: app/assets/stylesheets/partials/_posts.scss (hunk 1 above)
Comment (mi-wood) [inline app/assets/stylesheets/partials/_posts.scss]: I guess it's just to show you added spaces...

File: spec/features/post_spec.rb [hunk 3]
```diff
@@ -135,3 +135,25 @@
...
+    expect(page).to_not have_content(post.street)
+  end
+
+  it "shows map" do
+    post = create(:post)
+
+    visit post_url(id: post.id)
+
+    expect(page).to have_css("#map")
+    # We should use a JS enabled driver, so we can check if mapkit loads
```
Comment (mi-wood) [inline spec/features/post_spec.rb]: should we link this to a TODO issue?

**Comment by mi-wood** [comment]
LGTM :heart: :apple: :green_apple: :green_heart:

File: spec/features/post_spec.rb (hunk 3 above)
Comment (cllns) [inline spec/features/post_spec.rb]: I just changed 'note' to TODO, since that'll be easier to search

**Comment by mi-wood** [comment]
query limits *sigh*. gotta create a mock for the responses :/

**Pull request merged**